framework/deproxy_server.py

```diff
@@ -128,18 +185,85 @@ def wait_for_connections(self, timeout=1):
             time.sleep(0.001) # to prevent redundant CPU usage
         return True
 
-    @abc.abstractmethod
-    def recieve_request(self, request, connection):
-        raise NotImplementedError("Not implemented 'recieve_request()'")
 
-class StaticDeproxyServer(BaseDeproxyServer):
+class StaticDeproxyServer(BaseAsyncoreDeproxyServer):
 
     def __init__(self, *args, **kwargs):
         self.response = kwargs['response']
         kwargs.pop('response', None)
-        BaseDeproxyServer.__init__(self, *args, **kwargs)
+        BaseAsyncoreDeproxyServer.__init__(self, *args, **kwargs)
         self.last_request = None
 
+    def init_socket(self):
+        self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
+
     def recieve_request(self, request, connection):
         self.last_request = request
         return deproxy.Response(self.response)
+
+class SSLStaticDeproxyServer(BaseAsyncoreDeproxyServer):
+
+    def __init__(self, cert, cert_key, *args, **kwargs):
+        self.response = kwargs['response']
+        self.cert = cert
+        self.cert_key = cert_key
+        kwargs.pop('response', None)
+        BaseAsyncoreDeproxyServer.__init__(self, *args, **kwargs)
+        self.last_request = None
+
+    def init_socket(self):
+        s = socket.socket()
+        self.ssl_socket = ssl.SSLSocket(sock=s, certfile=self.cert, keyfile=self.cert_key)
+        self.set_socket(self.ssl_socket)
+
+    def recieve_request(self, request, connection):
+        self.last_request = request
+        return deproxy.Response(self.response)
+
+def create_srv_deproxy(server):
+    ka = "close"
+    if server.has_key('connection_header'):
+        ka = server['connection_header']
+    port = server['port']
+    if port == 'default':
+        port = tempesta.upstream_port_start_from()
+    else:
+        port = int(port)
+    srv = None
+    rtype = server['response']
+    if rtype == 'static':
+        content = fill_template(server['response_content'])
+        srv = StaticDeproxyServer(listen_port=port,
+                                  response=content,
+                                  keep_alive=ka,
+                                  conns_n=None)
+    else:
+        raise Exception("Invalid response type: %s" % str(rtype))
+
+    return srv
+
+def create_srv_deproxy_ssl(server):
```

Comment:
this is different functions because they creates different classes. I've used 1 class - 1 creation function method

Reply:
I just asked because the  functions look very close. I was expecting to see fabric pattern here: single function which creates the right class according to it's description.  But it's a part of `__create_client()` function, which is the fabric actually, i missed that, sorry.